src/index.js

```diff
@@ -1,45 +1,154 @@
+import SubscribeResponseV1 from './protocol/control_layer/subscribe_response/SubscribeResponseV1'
+
+import UnicastMessage from './protocol/control_layer/unicast_message/UnicastMessage'
+import UnicastMessageFactory from './protocol/control_layer/unicast_message/UnicastMessageFactory'
+import UnicastMessageV0 from './protocol/control_layer/unicast_message/UnicastMessageV0'
+import UnicastMessageV1 from './protocol/control_layer/unicast_message/UnicastMessageV1'
+
+import UnsubscribeRequest from './protocol/control_layer/unsubscribe_request/UnsubscribeRequest'
+import UnsubscribeRequestFactory from './protocol/control_layer/unsubscribe_request/UnsubscribeRequestFactory'
+import UnsubscribeRequestV0 from './protocol/control_layer/unsubscribe_request/UnsubscribeRequestV0'
+import UnsubscribeRequestV1 from './protocol/control_layer/unsubscribe_request/UnsubscribeRequestV1'
+
+import UnsubscribeResponse from './protocol/control_layer/unsubscribe_response/UnsubscribeResponse'
+import UnsubscribeResponseFactory from './protocol/control_layer/unsubscribe_response/UnsubscribeResponseFactory'
+import UnsubscribeResponseV0 from './protocol/control_layer/unsubscribe_response/UnsubscribeResponseV0'
+import UnsubscribeResponseV1 from './protocol/control_layer/unsubscribe_response/UnsubscribeResponseV1'
+
+import ControlMessage from './protocol/control_layer/ControlMessage'
+import ControlMessageFactory from './protocol/control_layer/ControlMessageFactory'
+import ControlMessageV0Factory from './protocol/control_layer/ControlMessageV0Factory'
+import ResendResponsePayload from './protocol/control_layer/ResendResponsePayload'
+import StreamAndPartition from './protocol/control_layer/StreamAndPartition'
+
+import MessageID from './protocol/message_layer/MessageID'
+import MessageRef from './protocol/message_layer/MessageRef'
+import StreamMessage from './protocol/message_layer/StreamMessage'
+import StreamMessageFactory from './protocol/message_layer/StreamMessageFactory'
+import StreamMessageV28 from './protocol/message_layer/StreamMessageV28'
+import StreamMessageV29 from './protocol/message_layer/StreamMessageV29'
+import StreamMessageV30 from './protocol/message_layer/StreamMessageV30'
+
 import InvalidJsonError from './errors/InvalidJsonError'
 import UnsupportedVersionError from './errors/UnsupportedVersionError'
-
 import TimestampUtil from './utils/TimestampUtil'
 
 module.exports = {
-    WebsocketRequest,
-    WebsocketResponse,
-    StreamMessage,
-    BroadcastMessage,
-    UnicastMessage,
-    PublishRequest,
-    ErrorResponse,
-    ErrorPayload,
-    ResendRequest,
-    ResendResponseNoResend,
-    ResendResponseResending,
-    ResendResponseResent,
-    ResendResponsePayload,
-    SubscribeRequest,
-    SubscribeResponse,
-    UnsubscribeRequest,
-    UnsubscribeResponse,
-    StreamAndPartition,
+    ControlLayer: {
+        BroadcastMessage,
+        BroadcastMessageFactory,
+        BroadcastMessageV0,
```

Comment:
Referring to our discussion: we could clean the API by exposing only what is needed. Ideally the whole `V0` and `V1` business would be hidden. Cleanest API would IMO be:

For creating:
`BroadcastMessage.create(...) // creates newest version, because why not`

For serializing:
`msg.serialize(version) // version defaults to newest version`

For deserializing a message of unknown type:
`ControlMessageFactory.deserialize(...)`

That would mean that only the following need to be imported and exposed here:
- Abstract top level message classes
- `ControlMessageFactory`